### PR TITLE
Automate draft release creation with Goreleaser

### PR DESCRIPTION
**Pull request by doodlesbykumbi**
### What does this PR do?

Adds release automation to Jenkins. Go releaser builds the release artefacts and creates a draft release on Github. Add apk to release artefacts.

### What ticket does this PR close?
Resolves -
Works towards https://github.com/cyberark/summon/issues/209

#### Change log
- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [ ] This PR includes new unit and integration tests to go with the code changes, or
- [x] The changes in this PR do not require tests

#### Documentation
- [x] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## Unreleased
 
+### Added
+- apk package
```
Comment (BradleyBoutcher) [inline CHANGELOG.md:10]: Could we add more detail here?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -91,19 +91,19 @@ The following checklist should be followed when creating a release:
...
-      - [ ] Attach `dist/summon_v*.deb` to release.
-      - [ ] Attach `dist/CHANGELOG.md` to release.
-      - [ ] Attach `dist/SHA256SUMS.txt` to the release.
+      - [ ] Attach `dist/goreleaser/summon-darwin-amd64.tar.gz` to release.
+      - [ ] Attach `dist/goreleaser/summon-linux-amd64.tar.gz` to release.
+      - [ ] Attach `dist/goreleaser/summon-windows-amd64.tar.gz` to release.
+      - [ ] Attach `dist/goreleaser/summon_v*.rpm` to release.
+      - [ ] Attach `dist/goreleaser/summon_v*.deb` to release.
+      - [ ] Attach `dist/goreleaser/CHANGELOG.md` to release.
+      - [ ] Attach `dist/goreleaser/SHA256SUMS.txt` to the release.
```
Comment (BradleyBoutcher) [inline CONTRIBUTING.md:100]: Question: Why do we need a `goreleaser` subdirectory? Are there other ways we'll build the binaries in the future, or is this just the default output directory of goreleaser?

**Review by BradleyBoutcher — COMMENTED** [review]
Looks great! Two small comments, but I think this is good to go

File: CONTRIBUTING.md (hunk 2 above)
Comment (doodlesbykumbi) [reply CONTRIBUTING.md:100]: Consistency more than anything. summon-conjur and another project are using the subdir

File: CHANGELOG.md (hunk 1 above)
Comment (doodlesbykumbi) [reply CHANGELOG.md]: Sorted

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## Unreleased
 
+### Added
+- Adds apk package to the release artefacts.
```
Comment (BradleyBoutcher) [inline CHANGELOG.md:10]: s/artefacts/artifacts

**Review by BradleyBoutcher — APPROVED** [review]

**Pull request merged**